Update websockify.js

brugnara opened the pull request:
Hi.

First of all, thank you for your great module.

I'm opening a request since I think declaring `argv` with a `var` in that point, breaks the other part of the code in where `argv` is also used because of the scope. Maybe an old version of node or without `strict` mode enabled this could works but it is not with node LTS, which is 8 at the time of writing.

In my case I found that passing the `.web` option, the line 79 breaks everything, since `argv` is `null` due to missing initialization, so `argv.web` fails accordingly.

Again, thank you.

Let me know what you think about this.

#3

maximegris commented:
Removing `var` defore `argv` declaration is fine. But why did you add `serveStatic` to this PR ? 
I fixed the master branch.

brugnara commented:
I just checked my pr and you should merge all of the code I added. I forgot
about I already have pushed the code with the servestatic module. This way
all of the files will be served correctly with the right mime type. Don't
stop checking the index.htm file. Js or images will not work in a complex
page with the original module. Let me know if you need more details or
examples.



Il dom 2 set 2018, 10:01 Daniele Brugnara <daniele@brugnara.me> ha scritto:

> This pr should not contain that dep. But the way static files are served
> in the original module is not working. Using the serve static module solves
> this. I'll put the workinig code tomorrow.
>
> Il dom 2 set 2018, 09:40 Maxime GRIS <notifications@github.com> ha
> scritto:
>
>> Removing var defore argv declaration is fine. But why did you add
>> serveStatic to this PR ?
>> I fixed the master branch.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/maximegris/node-websockify/pull/6#issuecomment-417911438>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ACpwByp30FR5x3e375CMHuWeDCfky8VEks5uW4txgaJpZM4WVVdp>
>> .
>>
> --
> Daniele Brugnara
>
-- 
Daniele Brugnara

brugnara commented on websockify.js:
```diff
@@ -80,31 +81,11 @@ var http_request = function (request, response) {
-			}
-
-			response.writeHead(200);
-			response.write(file, "binary");
```
@maximegris this causes:

```
Refused to execute script from 'data.min.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.
```

Using serve-static module the problem is solved.
